### PR TITLE
feat: support aten.copy dynamo converter

### LINKED ISSUE #2435: Add support for `aten.copy`

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

Support `aten.copy` dynamo converter.

Fixes #2435 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 1]
```diff
@@ -2541,3 +2541,28 @@ def aten_ops_sort(
         dim=args_bounds_check(args, 1, -1),
         descending=args_bounds_check(args, 2, False),
     )
+
+
+@dynamo_tensorrt_converter(torch.ops.aten.copy.default)
+@enforce_tensor_types(
+    {
+        0: (TRTTensor,),
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2549]: Should this be `1: (TRTTensor,)`? We don't seem to use the first argument in the converter

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 2]
```diff
@@ -2541,3 +2541,28 @@ def aten_ops_sort(
...
+    src = args[1]
+    return impl.cast.to_copy(
+        ctx,
+        target,
+        SourceIR.ATEN,
+        name,
+        src,
+        src.dtype,
+        force_layer=True,
+    )
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2568]: The `force_layer=True` does seem useful for the purpose of the test case, but in a normal model we should prefer `force_layer=False`. This should be a future feature, but `to_copy` should have an intelligent `force_layer` mechanism which can select whether to insert the layer based on whether the node is an `input` node to the TRTEngine.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 1 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2549]: Thanks for the good catch!

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 2 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2568]: Yes, I agree with you, but for now TensorRT doesn't support the same input and output.

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2574]: I added #2561 to track this proposed feature (in Torch-TRT)

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 3]
```diff
@@ -2530,7 +2530,7 @@ def aten_ops_sort(
 @dynamo_tensorrt_converter(torch.ops.aten.trunc.default)
 @enforce_tensor_types(
     {
-        0: (TRTTensor,),
+        1: (TRTTensor,),
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2533]: It doesn't seem that this modification should apply to `trunc` here

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2533]: Oh good catch! That was accidentally involved by rebase.

**Review by gs-olive — APPROVED** [review]
Looks good to me!

**Pull request merged**